File: Behavioral/Lab21/Typescript/src/Entities/person.ts
import { AgencyObject } from './agency_object';

export abstract class Person extends AgencyObject {
    public nameAndSurname: string;
    public phone: string;
    public email: string;
    public location: string = "Ukraine";

    constructor();
    constructor(nameAndSurname: string, phone: string, email: string);
    constructor(nameAndSurname?: string, phone?: string, email?: string) {
        super();
        if (nameAndSurname && phone && email) {
            this.nameAndSurname = nameAndSurname;
            this.phone = phone;
            this.email = email;
        } 
    }

    // protected createDefault():Person {
    //     return new Person("Jane Doe", "8 800 555 35 35", "example@gmail.com");
    // }
}

File: Creational/Lab5/C#/Person.cs
namespace C_
{
    public abstract class Person : AgencyObject
    {
        public string NameAndSurname { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }

        public Person(string nameAndSurname, string phone, string email):base() { 
            NameAndSurname = nameAndSurname;
            Phone = phone;
            Email = email;
        }

        public Person():base(){
            NameAndSurname = "Jane Doe";
            Phone = "8 800 555 35 35";
            Email = "example@gmail.com";
        }
    }
}


